README.md

```diff
@@ -118,16 +108,48 @@ Fired when the scroller has finished rendering a new range of items, e.g. becaus
 
 Also see [the example below](#performing-actions-as-the-scroller-scrolls-using-the-rangechange-event).
 
+### The `ItemSource` class
+
+The `ItemSource` class represents a way of translating indices into JavaScript values. You can create them like so:
+
+```js
+const source = new ItemSource({
+  item(index) { ... },
+  getLength() { ... },
+  key(index) { ... }
+});
+```
+
+For example, to create an `ItemSource` that gets its items from a `contacts` array, and uses `contact.id` as the key, you could do
+
+```js
+const contactsSource = new ItemSource({
+  item(index) { return contacts[index]; },
+  getLength() { return contacts.length; },
+  key(index) { return contacts[index].id; }
+});
+```
+
+There is also a factory method, `ItemSource.fromArray(array[, key])`, that makes this easier:
+
+```js
+const contactsSource = ItemSource.fromArray(contacts, c => c.id);
+```
+
+The `key` argument to `fromArray()` is called with an item, and should return a unique key for the object. If no `key` argument is given, then the item itself is used as the key.
+
+The main use of the `ItemSource` class is to be assigned to the `itemSource` property of a `<virtual-scroller>` element; as such, for now its only public API is a `length` property.

```

Comment:
`ItemSource` should also expose `item, key` properties so that `virtual-scroller` can access to them.

Reply:
It can use the exported symbols, was my plan

Reply:
oh wow, I didn't even see they were exported! Awesome!